skbio/io/format/genbank.py

```diff
@@ -0,0 +1,669 @@
+
+
+@genbank.writer(RNA)
+def _rna_to_genbank(obj, fh, lowercase=None):
+    _serialize_single_genbank(obj, fh, lowercase)
+
+
+@genbank.writer(Protein)
+def _protein_to_genbank(obj, fh, lowercase=None):
+    _serialize_single_genbank(obj, fh, lowercase)
+
+
+def _construct(record, constructor=None, **kwargs):
+    seq, md, pmd = record
+    if 'lowercase' not in kwargs:
+        kwargs['lowercase'] = True
+    if constructor is None:
+        unit = md['LOCUS']['unit']
+        if unit == 'bp':
+            is_rna = 'u' in seq or 'U' in seq
+            # often RNA mol type has T instead of U for genbank from from NCBI;
+            # but in case of files from other sources...
+            constructor = RNA if is_rna else DNA
+        elif unit == 'aa':
+            constructor = Protein
+        else:
+            constructor = Sequence
+    return constructor(
+        seq, metadata=md, positional_metadata=pmd, **kwargs)
+
+
+def _parse_genbanks(fh):
+    data_chunks = []
+    for line in _line_generator(fh, skip_blanks=True, strip=False):
+        if line.startswith('//'):
+            yield _parse_single_genbank(data_chunks)
+            data_chunks = []
+        else:
+            data_chunks.append(line)
+
+
+def _parse_single_genbank(chunks):
+    metadata = {}
+    positional_metadata = None
+    sequence = ''
+    # each section starts with a HEADER without indent.
+    section_splitter = yield_section(lambda x: not x[0].isspace(), strip=False)
+    for section in section_splitter(chunks):
+        header = section[0].split(None, 1)[0]
+        parser = _PARSER_TABLE.get(
+            header, _parse_section_default)
+
+        if header == 'FEATURES':
+            parser = partial(
+                parser, length=metadata['LOCUS']['size'])
+
+        try:
+            parsed = parser(section)
+        except:
+            raise GenbankFormatError(
+                'Could not parse this section with %s:\n%s' %
+                (parser, ''.join(section)))
+
+        # reference can appear multiple times
+        if header == 'REFERENCE':
+            if header in metadata:
+                metadata[header].append(parsed)
+            else:
+                metadata[header] = [parsed]
+        elif header == 'ORIGIN':
+            sequence = parsed
+        elif header == 'FEATURES':
+            metadata[header] = parsed[0]
+            positional_metadata = pd.concat(parsed[1], axis=1)
+        else:
+            metadata[header] = parsed
+    return sequence, metadata, positional_metadata
+
+
+def _serialize_single_genbank(obj, fh, lowercase=None):
+    md = obj.metadata
+    for header in _HEADERS:
+        if header in md:
+            serializer = _SERIALIZER_TABLE.get(
+                header, _serialize_section_default)
+            out = serializer(header, md[header])
+            # test if 'out' is a iterator.
+            # cf. Effective Python Item 17
+            if iter(out) is iter(out):
```

Comment:
not if `a` is an iterator. try:

````
b = iter(a)
iter(b) is iter(b)
````

The `iter` protocol states that `iter()` will always return the iterator itself.